docs/pyfdb/examples.rst

```diff
@@ -0,0 +1,816 @@
+Example usage
+#############
+
+This example demonstrates how to use common functionality of the ``PyFDB``.
+
+PyFDB Intro
+***********
+
+.. code-block:: python
+
+    fdb_config_path = Path("<path-to-fdb-config>")
+    
+    fdb_config = pyfdb.Config(config_file.read_text())
+    pyfdb = pyfdb.PyFDB(fdb_config)
+
+   fdb_config_path = Path("<path-to-fdb-config>")
+   fdb_config = pyfdb.Config(config_file.read_text())
+   pyfdb = pyfdb.PyFDB(fdb_config)
+
+
+
+You can also pass custom configurations. Those can be supplied as a ``Path`` pointing to the location
+of the configuration file, as a ``str`` which is the yaml representation of the configuration or as
+a ``Dict[str, Any]`` as shown below. 
+
+.. code-block:: python
+
+    config = dict(
+        type="local",
+        engine="toc",
+        schema="/path/to/fdb_schema",
+        spaces=[
+            dict(
+                handler="Default",
+                roots=[
+                    {"path": "/path/to/root"}
+                ]
+            )
+        ],
+    )
+    fdb = pyfdb.PyFDB(config=config, userconfig={})
+
+.. TODO(TKR): Specify what a user config is and how this can be used here.
+
+
+
+
+
+
+Down below are some of the common examples you want to use `PyFDB` for:
+
+Archive
+***********
+
+**Archive binary data into the underlying FDB.**
+
+.. code-block:: python
+
+    fdb_config = pyfdb.Config(config_file.read_text())
+    pyfdb = pyfdb.PyFDB(fdb_config)
+
+    filename = test_data_path / "x138-300.grib"
+
+    pyfdb.archive(open(filename, "rb").read())
+    pyfdb.flush()
+
+In this scenario a ``GRIB`` file is archived to the configured ``FDB``. The FDB reads metadata from
+the given ``GRIB`` file and safes this, if no optional ``identifier`` is supplied. If we set an
+``identifier``, there are no consistency checks taking place and our data is saved with the metadata
+given from the supplied ``identifier``.
+
+The ``flush`` command guarantees that the archived data has been flushed to the ``FDB``.
```

Comment:
I think that this is under clear about flush. Flush indicates to the FDB that it _must_ flush the data, and blocks until it is complete.